platform/mbed_enum_flags.h

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (c) 2020 ARM Limited. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ * Licensed under the Apache License, Version 2.0 (the License); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MBED_SCOPED_ENUM_FLAGS_H
+#define MBED_SCOPED_ENUM_FLAGS_H
+
+#ifdef __cplusplus
+#include <type_traits>
+
+#define ENUM_FLAG_OPERATOR(T,OP) \
+inline constexpr T operator OP( T lhs, T rhs) {\
+    return ( T ) ( \
+        static_cast<std::underlying_type< T >::type>(lhs) OP \
+        static_cast<std::underlying_type< T >::type>(rhs)); \
+};
+
+
+#define ENUM_FLAG_OPERATOR_LHS_REFERNCE(T, X) \
+inline T & operator X (T & a, T b) { \
+    a = a X b; \
+    return a; \
```

Comment:
I like making as much as possible `constexpr`.

I'll blame it on my thinking that I was under the incorrect belief that mbed was now C++11.

I will shorten the function return statement to one line.

I also changed the function style to put the opening brace on the line after the function definition.

Reply:
One more comment for this one: I also will change `X` in this implementation to `OP` like the other definition. Even though `X` makes some amount of sense to me, I like `OP` more because it is a little more obvious that it is an operator and not a generic placename.